crates/sncast/src/main.rs

```diff
@@ -66,21 +67,21 @@ Report bugs: https://github.com/foundry-rs/starknet-foundry/issues/new/choose\
 #[allow(clippy::struct_excessive_bools)]
 struct Cli {
     /// Profile name in snfoundry.toml config file
-    #[clap(short, long)]
+    #[clap(short, long, global = true)]
```

Comment:
Instead of making these arguments global, let's actually add them as arguments to subcommands that use them. And keep them as global as well to retain current cli.

If a user uses the global version show a warning but continue program execution.

If both are provided, throw an error.